### ISSUE TITLE
Distributed tests fail 

### ISSUE THREAD

**Issue by KickItLikeShika** [pytorch/ignite#1792]
Recently while working on `ignite.contrib.metrics.regression` tests, I was trying to add distributed tests to some metrics that didn't have distributed tests. While i was running those tests when `world_size > 1` the result computed by the metric implemented in `ignite` differs from the one computed by `sklearn`(if imported from sklearn) / `numpy`. And here are examples for that:

- [Test wave hedges distance](https://gist.github.com/KickItLikeShika/497e80b2ebf2b99a9775e095868b62af)
`_test_distrib_integration()` fails in case of `world_size > 1` and if i didn't comment `all_gather()` in `_test_distrib_compute()` it fails as well!

- [Test fractional absolute error](https://gist.github.com/KickItLikeShika/8b6919f08b4d5ba531a363580517f791)
The same failures in the same two tests `_test_distrib_compute()` and `_test_distrib_integration()`.

related: #1284

**Comment by sdesrozis** [comment]
If I understand correctly the test, every process generates random numbers and compute the metric. Then, the result is compared to a score computed with numpy. But you have to uncomment the gather part otherwise it can’t be similar.

**Comment by KickItLikeShika** [comment]
@sdesrozis Thanks for your comment! You understand the test correctly. I'm already doing `all_gather()` and i know it should be done, but the point is the `_test_distrib_compute()` doesn't work if i did `all_gather()`. And `_test_distrib_integration()` doesn't work no matter i'm gathering or not. Btw, the same problem happens with many other tests, for example: `test_mean_...`, `test_fractional_bias`. And i will try the other tests, it mostly will fail the same way...

**Comment by KickItLikeShika** [comment]
I don't know if it's related, but i see a weird pattern here. Dor metrics implemented in `sklearn` and we import `sklearn` to use it for computing the metric it works fine, I guess the problem is with metrics we computed it with `numpy`

**Comment by sdesrozis** [comment]
I will have a look asap

**Comment by sdesrozis** [comment]
Ah ah take a look, the metrics are not distributed. It means a reduction is needed in the implementation. I could fix it tomorrow.

**Comment by KickItLikeShika** [comment]
@sdesrozis That's great to hear! If you provided me some hints about what should be done i will be happy to help

**Comment by vfdev-5** [comment]
@sdesrozis I think we can leave this task to @KickItLikeShika . 

@KickItLikeShika check here: https://pytorch.org/ignite/metrics.html#metrics-and-distributed-computations
and here : https://pytorch.org/ignite/distributed.html#ignite-distributed-utils for the list of collective ops.
Please, try to figure out how it works by your own and ask if you tried everything and you become blocked for sometime. Thanks !
